src/equality.rs

```diff
@@ -12,24 +12,42 @@ pub struct Equality<L: SynthLanguage> {
     pub rewrite: Rewrite<L, SynthAnalysis>,
 }
 
-impl<L: SynthLanguage> FromStr for Equality<L> {
-    type Err = String;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+impl<L: SynthLanguage> Equality<L> {
+    pub fn from_string(s: &str) -> Result<(Self, Option<Self>), String> {
         if let Some((l, r)) = s.split_once("=>") {
             let l_pat: Pattern<L> = l.parse().unwrap();
             let r_pat: Pattern<L> = r.parse().unwrap();
```

Comment:
I think we need to settle on a uniform unidirectional rules format. If we have `a ==> b` and `a => b` in the repo, I'm not sure if this will parse correctly. Separate note, maybe we should try to split on `<=>` first and then do the unidirectional one.